File: Examenes/ExamenPSP/2DAM_PSP-main/11 - Spring GraphQL/graphQLServer/src/main/java/com/serverql/graphqlserver/domain/modelo/Mascota.java
package com.serverql.graphqlserver.domain.modelo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;



@Data
@AllArgsConstructor
@NoArgsConstructor
public class Mascota {

    private int id;

    private String name;

    private String type;
    private int age;

    private int persona;
}

File: Examenes/ExamenPSP/2DAM_PSP-main/11 - Spring GraphQL/graphqlCliente/src/main/kotlin/domain/modelo/Mascota.kt
package domain.modelo

data class Mascota(var id: Int = 0, var name: String = "", val age: Int = 0, val type: String = "", val persona: Persona = Persona()){
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Mascota

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id
    }
}